cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDownloadByteChannel.java

```diff
@@ -0,0 +1,43 @@
+package mil.army.usace.hec.cumulus.client.controllers;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.util.function.IntConsumer;
+
+final class CumulusDownloadByteChannel implements ReadableByteChannel {
+
+    private final ReadableByteChannel readableByteChannel;
+    private final IntConsumer bytesReadConsumer;
+    private int bytesReadSoFar;
+
+    CumulusDownloadByteChannel(ReadableByteChannel readableByteChannel, IntConsumer bytesReadConsumer) {
+        this.readableByteChannel = readableByteChannel;
+        this.bytesReadConsumer = bytesReadConsumer;
+    }
+
+    @Override
+    public int read(ByteBuffer byteBuffer) throws IOException {
+        int newBytesRead = readableByteChannel.read(byteBuffer);
+        notifyBytesRead(newBytesRead);
+        return newBytesRead;
+    }
+
+    private void notifyBytesRead(int bytesRead) {
+        if (bytesRead <= 0) {
+            return;
+        }
+        bytesReadSoFar += bytesRead;
+        bytesReadConsumer.accept(bytesReadSoFar);
```

Comment:
this should be current and total

Reply:
Do we have a way of getting the total?

Reply:
Updated to notify current bytes being read, and total bytes read so far